src/Workspaces/Core/Portable/Workspace/Host/Metadata/IAnalyzerAssemblyLoaderProvider.cs

```diff
@@ -35,30 +35,60 @@ internal interface IAnalyzerAssemblyLoaderProvider : IWorkspaceService
 /// </summary>
 internal abstract class AbstractAnalyzerAssemblyLoaderProvider : IAnalyzerAssemblyLoaderProvider
 {
-    private readonly ImmutableArray<IAnalyzerAssemblyResolver> _externalResolvers;
+#if NET
     private readonly Lazy<IAnalyzerAssemblyLoaderInternal> _shadowCopyLoader;
+    private readonly ImmutableArray<IAnalyzerAssemblyResolver> _assemblyResolvers;
 
-    public AbstractAnalyzerAssemblyLoaderProvider(IEnumerable<IAnalyzerAssemblyResolver> externalResolvers)
+    public AbstractAnalyzerAssemblyLoaderProvider(IEnumerable<IAnalyzerAssemblyResolver> assemblyResolvers)
     {
-        _externalResolvers = [.. externalResolvers];
+        _assemblyResolvers = [.. assemblyResolvers];
         _shadowCopyLoader = new(CreateNewShadowCopyLoader);
     }
 
     public IAnalyzerAssemblyLoaderInternal SharedShadowCopyLoader
         => _shadowCopyLoader.Value;
 
     public IAnalyzerAssemblyLoaderInternal CreateNewShadowCopyLoader()
-        => this.WrapLoader(DefaultAnalyzerAssemblyLoader.CreateNonLockingLoader(
+        => this.WrapLoader(AnalyzerAssemblyLoader.CreateNonLockingLoader(
                 Path.Combine(Path.GetTempPath(), nameof(Roslyn), "AnalyzerAssemblyLoader"),
-                _externalResolvers));
+                pathResolvers: default,
+                _assemblyResolvers));
+#else
+    private readonly Lazy<IAnalyzerAssemblyLoaderInternal> _shadowCopyLoader;
+
+    public AbstractAnalyzerAssemblyLoaderProvider()
+    {
+        _shadowCopyLoader = new(CreateNewShadowCopyLoader);
+    }
+
+    public IAnalyzerAssemblyLoaderInternal SharedShadowCopyLoader
+        => _shadowCopyLoader.Value;
+
+    public IAnalyzerAssemblyLoaderInternal CreateNewShadowCopyLoader()
+        => this.WrapLoader(AnalyzerAssemblyLoader.CreateNonLockingLoader(
+                Path.Combine(Path.GetTempPath(), nameof(Roslyn), "AnalyzerAssemblyLoader"),
+                pathResolvers: default));
+#endif
 
     protected virtual IAnalyzerAssemblyLoaderInternal WrapLoader(IAnalyzerAssemblyLoaderInternal loader)
         => loader;
 }
 
 [ExportWorkspaceService(typeof(IAnalyzerAssemblyLoaderProvider)), Shared]
-[method: ImportingConstructor]
-[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-internal sealed class DefaultAnalyzerAssemblyLoaderProvider(
-    [ImportMany] IEnumerable<IAnalyzerAssemblyResolver> externalResolvers)
-    : AbstractAnalyzerAssemblyLoaderProvider(externalResolvers);
+internal sealed class DefaultAnalyzerAssemblyLoaderProvider : AbstractAnalyzerAssemblyLoaderProvider
```

Comment:
Moving `IAnalyzerAssemblyResolver` to be .NET Core only was correct but I hate the bifuraction it caused in the MEF layer here. I did give consideration for adding a new factory interface that had a `#if NET` method for producing this value. That would simplify the MEF import here but eventually decided to go this route. Happy to change if you all feel differently.

@jasonmalinowski

Reply:
I'll think about this more next week if there's some nicer alternative, but it's constrained enough here to not be too much of a mess.